with grade 2.5: shadowJar task fails, Caused by: java.lang.NoClassDefFoundError

```
Caused by: java.lang.NoClassDefFoundError: org.gradle.mvn3.org.codehaus.plexus.util.IOUtil
	at com.github.jengelman.gradle.plugins.shadow.transformers.ServiceFileTransformer$ServiceStream.class$(ServiceFileTransformer.groovy)
	at com.github.jengelman.gradle.plugins.shadow.transformers.ServiceFileTransformer$ServiceStream.$get$$class$org$gradle$mvn3$org$codehaus$plexus$util$IOUtil(ServiceFileTransformer.groovy)
	at com.github.jengelman.gradle.plugins.shadow.transformers.ServiceFileTransformer$ServiceStream.append(ServiceFileTransformer.groovy:98)
	at com.github.jengelman.gradle.plugins.shadow.transformers.ServiceFileTransformer$ServiceStream$append.call(Unknown Source)
	at com.github.jengelman.gradle.plugins.shadow.transformers.ServiceFileTransformer.transform(ServiceFileTransformer.groovy:70)
	at com.github.jengelman.gradle.plugins.shadow.transformers.Transformer$transform$0.call(Unknown Source)
	at com.github.jengelman.gradle.plugins.shadow.tasks.ShadowCopyAction$StreamAction.transform(ShadowCopyAction.groovy:323)
	at com.github.jengelman.gradle.plugins.shadow.tasks.ShadowCopyAction$StreamAction.this$2$transform(ShadowCopyAction.groovy)
	at com.github.jengelman.gradle.plugins.shadow.tasks.ShadowCopyAction$StreamAction$this$2$transform$9.callCurrent(Unknown Source)
	at com.github.jengelman.gradle.plugins.shadow.tasks.ShadowCopyAction$StreamAction.transform(ShadowCopyAction.groovy:314)
	at com.github.jengelman.gradle.plugins.shadow.tasks.ShadowCopyAction$StreamAction.this$2$transform(ShadowCopyAction.groovy)
	at com.github.jengelman.gradle.plugins.shadow.tasks.ShadowCopyAction$StreamAction$this$2$transform$8.callCurrent(Unknown Source)
	at com.github.jengelman.gradle.plugins.shadow.tasks.ShadowCopyAction$StreamAction.visitArchiveFile(ShadowCopyAction.groovy:229)
	at com.github.jengelman.gradle.plugins.shadow.tasks.ShadowCopyAction$StreamAction.this$2$visitArchiveFile(ShadowCopyAction.groovy)
	at com.github.jengelman.gradle.plugins.shadow.tasks.ShadowCopyAction$_StreamAction_processArchive_closure3.doCall(ShadowCopyAction.groovy:204)
	at com.github.jengelman.gradle.plugins.shadow.tasks.ShadowCopyAction$StreamAction.processArchive(ShadowCopyAction.groovy:202)
	at com.github.jengelman.gradle.plugins.shadow.tasks.ShadowCopyAction$StreamAction.this$2$processArchive(ShadowCopyAction.groovy)
	at com.github.jengelman.gradle.plugins.shadow.tasks.ShadowCopyAction$StreamAction$this$2$processArchive$4.callCurrent(Unknown Source)
	at com.github.jengelman.gradle.plugins.shadow.tasks.ShadowCopyAction$StreamAction.visitFile(ShadowCopyAction.groovy:188)
	at com.github.jengelman.gradle.plugins.shadow.tasks.ShadowCopyAction$StreamAction.this$2$visitFile(ShadowCopyAction.groovy)
	at com.github.jengelman.gradle.plugins.shadow.tasks.ShadowCopyAction$StreamAction$this$2$visitFile$1.callCurrent(Unknown Source)
	at com.github.jengelman.gradle.plugins.shadow.tasks.ShadowCopyAction$StreamAction.processFile(ShadowCopyAction.groovy:151)
	at org.gradle.api.internal.file.copy.NormalizingCopyActionDecorator$1$1.processFile(NormalizingCopyActionDecorator.java:65)
	at org.gradle.api.internal.file.copy.DuplicateHandlingCopyActionDecorator$1$1.processFile(DuplicateHandlingCopyActionDecorator.java:60)
	at org.gradle.api.internal.file.copy.CopyFileVisitorImpl.processFile(CopyFileVisitorImpl.java:60)
	at org.gradle.api.internal.file.copy.CopyFileVisitorImpl.visitFile(CopyFileVisitorImpl.java:44)
	at org.gradle.api.internal.file.AbstractFileTree$FilteredFileTree$1.visitFile(AbstractFileTree.java:146)
	at org.gradle.api.internal.file.collections.SingletonFileTree.visit(SingletonFileTree.java:44)
	at org.gradle.api.internal.file.collections.FileTreeAdapter.visit(FileTreeAdapter.java:108)
	at org.gradle.api.internal.file.AbstractFileTree$FilteredFileTree.visit(AbstractFileTree.java:137)
	at org.gradle.api.internal.file.CompositeFileTree.visit(CompositeFileTree.java:55)
	at org.gradle.api.internal.file.copy.CopySpecActionImpl.execute(CopySpecActionImpl.java:37)
	at org.gradle.api.internal.file.copy.CopySpecActionImpl.execute(CopySpecActionImpl.java:24)
	at org.gradle.api.internal.file.copy.DefaultCopySpec$DefaultCopySpecResolver.walk(DefaultCopySpec.java:498)
	at org.gradle.api.internal.file.copy.DefaultCopySpec$DefaultCopySpecResolver.walk(DefaultCopySpec.java:500)
	at org.gradle.api.internal.file.copy.DefaultCopySpec.walk(DefaultCopySpec.java:322)
	at org.gradle.api.internal.file.copy.CopySpecBackedCopyActionProcessingStream.process(CopySpecBackedCopyActionProcessingStream.java:36)
	at org.gradle.api.internal.file.copy.DuplicateHandlingCopyActionDecorator$1.process(DuplicateHandlingCopyActionDecorator.java:44)
	at org.gradle.api.internal.file.copy.NormalizingCopyActionDecorator$1.process(NormalizingCopyActionDecorator.java:56)
	at org.gradle.api.internal.file.copy.CopyActionProcessingStream$process.call(Unknown Source)
	at com.github.jengelman.gradle.plugins.shadow.tasks.ShadowCopyAction$1.execute(ShadowCopyAction.groovy:78)
	at com.github.jengelman.gradle.plugins.shadow.tasks.ShadowCopyAction$1$execute.call(Unknown Source)
	at com.github.jengelman.gradle.plugins.shadow.tasks.ShadowCopyAction.withResource(ShadowCopyAction.groovy:109)
	at com.github.jengelman.gradle.plugins.shadow.tasks.ShadowCopyAction.execute(ShadowCopyAction.groovy:75)
	at org.gradle.api.internal.file.copy.NormalizingCopyActionDecorator.execute(NormalizingCopyActionDecorator.java:52)
	at org.gradle.api.internal.file.copy.DuplicateHandlingCopyActionDecorator.execute(DuplicateHandlingCopyActionDecorator.java:42)
	at org.gradle.api.internal.file.copy.CopyActionExecuter.execute(CopyActionExecuter.java:38)
	at org.gradle.api.tasks.AbstractCopyTask.copy(AbstractCopyTask.java:83)
	at com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar.copy(ShadowJar.java:64)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:75)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.doExecute(AnnotationProcessingTaskFactory.java:226)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.execute(AnnotationProcessingTaskFactory.java:219)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.execute(AnnotationProcessingTaskFactory.java:208)
	at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:585)
	at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:568)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:80)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:61)
```

--- a/build.gradle
+++ b/build.gradle
@@ -44,6 +44,7 @@ dependencies {
     compile 'org.ow2.asm:asm-commons:5.0.3'
     compile 'commons-io:commons-io:2.4'
     compile 'org.apache.ant:ant:1.9.4'
+    compile 'org.codehaus.plexus:plexus-utils:2.0.6'
 
     testUtilsCompile localGroovy()
     testUtilsCompile gradleApi()

--- a/src/main/groovy/com/github/jengelman/gradle/plugins/shadow/relocation/SimpleRelocator.groovy
+++ b/src/main/groovy/com/github/jengelman/gradle/plugins/shadow/relocation/SimpleRelocator.groovy
@@ -19,7 +19,7 @@
 
 package com.github.jengelman.gradle.plugins.shadow.relocation
 
-import org.gradle.mvn3.org.codehaus.plexus.util.SelectorUtils
+import org.codehaus.plexus.util.SelectorUtils
 
 import java.util.regex.Pattern
 

--- a/src/main/groovy/com/github/jengelman/gradle/plugins/shadow/transformers/ApacheNoticeResourceTransformer.groovy
+++ b/src/main/groovy/com/github/jengelman/gradle/plugins/shadow/transformers/ApacheNoticeResourceTransformer.groovy
@@ -22,8 +22,8 @@ package com.github.jengelman.gradle.plugins.shadow.transformers
 import com.github.jengelman.gradle.plugins.shadow.relocation.Relocator
 import org.apache.tools.zip.ZipEntry
 import org.apache.tools.zip.ZipOutputStream
+import org.codehaus.plexus.util.StringUtils
 import org.gradle.api.file.FileTreeElement
-import org.gradle.mvn3.org.codehaus.plexus.util.StringUtils
 
 import java.text.SimpleDateFormat
 

--- a/src/main/groovy/com/github/jengelman/gradle/plugins/shadow/transformers/AppendingTransformer.groovy
+++ b/src/main/groovy/com/github/jengelman/gradle/plugins/shadow/transformers/AppendingTransformer.groovy
@@ -22,8 +22,8 @@ package com.github.jengelman.gradle.plugins.shadow.transformers
 import com.github.jengelman.gradle.plugins.shadow.relocation.Relocator
 import org.apache.tools.zip.ZipEntry
 import org.apache.tools.zip.ZipOutputStream
+import org.codehaus.plexus.util.IOUtil
 import org.gradle.api.file.FileTreeElement
-import org.gradle.mvn3.org.codehaus.plexus.util.IOUtil
 
 /**
  * A resource processor that appends content for a resource, separated by a newline.

--- a/src/main/groovy/com/github/jengelman/gradle/plugins/shadow/transformers/ComponentsXmlResourceTransformer.groovy
+++ b/src/main/groovy/com/github/jengelman/gradle/plugins/shadow/transformers/ComponentsXmlResourceTransformer.groovy
@@ -23,12 +23,12 @@ import com.github.jengelman.gradle.plugins.shadow.relocation.Relocator
 import org.apache.tools.zip.ZipEntry
 import org.apache.tools.zip.ZipOutputStream
 import org.gradle.api.file.FileTreeElement
-import org.gradle.mvn3.org.codehaus.plexus.util.IOUtil
-import org.gradle.mvn3.org.codehaus.plexus.util.ReaderFactory
-import org.gradle.mvn3.org.codehaus.plexus.util.WriterFactory
-import org.gradle.mvn3.org.codehaus.plexus.util.xml.Xpp3Dom
-import org.gradle.mvn3.org.codehaus.plexus.util.xml.Xpp3DomBuilder
-import org.gradle.mvn3.org.codehaus.plexus.util.xml.Xpp3DomWriter
+import org.codehaus.plexus.util.IOUtil
+import org.codehaus.plexus.util.ReaderFactory
+import org.codehaus.plexus.util.WriterFactory
+import org.codehaus.plexus.util.xml.Xpp3Dom
+import org.codehaus.plexus.util.xml.Xpp3DomBuilder
+import org.codehaus.plexus.util.xml.Xpp3DomWriter
 
 /**
  * A resource processor that aggregates plexus <code>components.xml</code> files.

--- a/src/main/groovy/com/github/jengelman/gradle/plugins/shadow/transformers/GroovyExtensionModuleTransformer.groovy
+++ b/src/main/groovy/com/github/jengelman/gradle/plugins/shadow/transformers/GroovyExtensionModuleTransformer.groovy
@@ -23,7 +23,7 @@ import com.github.jengelman.gradle.plugins.shadow.relocation.Relocator
 import org.apache.tools.zip.ZipEntry
 import org.apache.tools.zip.ZipOutputStream
 import org.gradle.api.file.FileTreeElement
-import org.gradle.mvn3.org.codehaus.plexus.util.IOUtil
+import org.codehaus.plexus.util.IOUtil
 
 /**
  * Modified from eu.appsatori.gradle.fatjar.tasks.PrepareFiles.groovy

--- a/src/main/groovy/com/github/jengelman/gradle/plugins/shadow/transformers/IncludeResourceTransformer.groovy
+++ b/src/main/groovy/com/github/jengelman/gradle/plugins/shadow/transformers/IncludeResourceTransformer.groovy
@@ -23,7 +23,7 @@ import com.github.jengelman.gradle.plugins.shadow.relocation.Relocator
 import org.apache.tools.zip.ZipEntry
 import org.apache.tools.zip.ZipOutputStream
 import org.gradle.api.file.FileTreeElement
-import org.gradle.mvn3.org.codehaus.plexus.util.IOUtil
+import org.codehaus.plexus.util.IOUtil
 
 /**
  * A resource processor that allows the addition of an arbitrary file

--- a/src/main/groovy/com/github/jengelman/gradle/plugins/shadow/transformers/ManifestResourceTransformer.groovy
+++ b/src/main/groovy/com/github/jengelman/gradle/plugins/shadow/transformers/ManifestResourceTransformer.groovy
@@ -23,7 +23,7 @@ import com.github.jengelman.gradle.plugins.shadow.relocation.Relocator
 import org.apache.tools.zip.ZipEntry
 import org.apache.tools.zip.ZipOutputStream
 import org.gradle.api.file.FileTreeElement
-import org.gradle.mvn3.org.codehaus.plexus.util.IOUtil
+import org.codehaus.plexus.util.IOUtil
 
 import java.util.jar.*
 import java.util.jar.Attributes.Name

--- a/src/main/groovy/com/github/jengelman/gradle/plugins/shadow/transformers/PropertiesFileTransformer.groovy
+++ b/src/main/groovy/com/github/jengelman/gradle/plugins/shadow/transformers/PropertiesFileTransformer.groovy
@@ -23,7 +23,7 @@ import com.github.jengelman.gradle.plugins.shadow.relocation.Relocator
 import org.apache.tools.zip.ZipEntry
 import org.apache.tools.zip.ZipOutputStream
 import org.gradle.api.file.FileTreeElement
-import org.gradle.mvn3.org.codehaus.plexus.util.IOUtil
+import org.codehaus.plexus.util.IOUtil
 
 /**
  * Resources transformer that merges Properties files.

--- a/src/main/groovy/com/github/jengelman/gradle/plugins/shadow/transformers/ServiceFileTransformer.groovy
+++ b/src/main/groovy/com/github/jengelman/gradle/plugins/shadow/transformers/ServiceFileTransformer.groovy
@@ -26,7 +26,7 @@ import org.gradle.api.file.FileTreeElement
 import org.gradle.api.specs.Spec
 import org.gradle.api.tasks.util.PatternFilterable
 import org.gradle.api.tasks.util.PatternSet
-import org.gradle.mvn3.org.codehaus.plexus.util.IOUtil
+import org.codehaus.plexus.util.IOUtil
 
 /**
  * Modified from org.apache.maven.plugins.shade.resource.ServiceResourceTransformer.java

--- a/src/test/groovy/com/github/jengelman/gradle/plugins/shadow/transformers/ComponentsXmlResourceTransformerTest.groovy
+++ b/src/test/groovy/com/github/jengelman/gradle/plugins/shadow/transformers/ComponentsXmlResourceTransformerTest.groovy
@@ -25,7 +25,7 @@ import org.custommonkey.xmlunit.Diff
 import org.custommonkey.xmlunit.XMLAssert
 import org.custommonkey.xmlunit.XMLUnit
 import com.github.jengelman.gradle.plugins.shadow.relocation.Relocator
-import org.gradle.mvn3.org.codehaus.plexus.util.IOUtil
+import org.codehaus.plexus.util.IOUtil
 
 /**
  * Test for {@link ComponentsXmlResourceTransformer}.

--- a/src/test/groovy/com/github/jengelman/gradle/plugins/shadow/util/JarBuilder.groovy
+++ b/src/test/groovy/com/github/jengelman/gradle/plugins/shadow/util/JarBuilder.groovy
@@ -1,6 +1,6 @@
 package com.github.jengelman.gradle.plugins.shadow.util
 
-import org.gradle.mvn3.org.codehaus.plexus.util.IOUtil
+import org.codehaus.plexus.util.IOUtil
 
 import java.util.jar.JarEntry
 import java.util.jar.JarOutputStream

--- a/src/test/groovy/com/github/jengelman/gradle/plugins/shadow/util/PluginSpecification.groovy
+++ b/src/test/groovy/com/github/jengelman/gradle/plugins/shadow/util/PluginSpecification.groovy
@@ -2,7 +2,7 @@ package com.github.jengelman.gradle.plugins.shadow.util
 
 import com.github.jengelman.gradle.testkit.GradleRunnerFactory
 import com.github.jengelman.gradle.testkit.file.TestFile
-import org.gradle.mvn3.org.codehaus.plexus.util.IOUtil
+import org.codehaus.plexus.util.IOUtil
 import org.gradle.testkit.functional.ExecutionResult
 import org.gradle.testkit.functional.GradleRunner
 import org.junit.Rule

